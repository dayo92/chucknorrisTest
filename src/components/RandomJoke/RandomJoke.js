import React, { useState} from 'react';
import Image from './img/parental.png';
import Img1 from './ranImg/tenor.gif';
import Img2 from './ranImg/tenor(1).gif';
import Img3 from './ranImg/tenor(2).gif';
import Img4 from './ranImg/tenor(3).gif';

 const RandomJoke = () => {
    const [jokes, setJokes] = useState('');
    const [currentImg, setCurrentImg] =useState([Img1, Img2, Img3, Img4])

    const getJoke=()=>{

        const timer = setTimeout(()=>{
        fetch("http://api.icndb.com/jokes/random/")
          .then(res => res.json())
          .then((result) => {
            
            setJokes(result.value.joke)
            },
    
          )
          .catch((err) => {
            console.log(err)
          });},1500)

          
          setJokes('')
            

      }
 
   
        const randomNumber = Math.floor(Math.random() * currentImg.length);



      return (
          
        <div className="wrapper">
            <h1>Chuck Norris Jokes</h1>
            <div className="randomJ">
                <div className="text">
                    {jokes ==='' ? <img alt="logo" src={Image}/> : <p className="quote" data-testid="get-joke">{`${jokes}`}</p>}
                </div>
            </div>
            <button className="button" onClick={getJoke}>get joke!</button>
            
            {jokes === '' ? null : <img className="norrisPic" alt='gif' src={currentImg[randomNumber]}/>}
            

           

        </div>
    );
}



export default RandomJoke;


