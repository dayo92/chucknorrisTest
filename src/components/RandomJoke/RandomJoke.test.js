import React from 'react';
import RandomJoke from './RandomJoke'
import {render, fireEvent, cleanup} from '@testing-library/react';


// it('API test', async function(){
//     const res = new RandomJoke; 
//     console.log( await res.getJoke)
// })

test('Snapshot of RandomJoke component', () => {
    const {debug, getByTestId} = render(
      
            <RandomJoke/>
      
    );
    expect(<RandomJoke/>).toMatchSnapshot()
     debug()
});

