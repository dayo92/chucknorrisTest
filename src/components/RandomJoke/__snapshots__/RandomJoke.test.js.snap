// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of RandomJoke component 1`] = `<RandomJoke />`;
